### PR TITLE
CR-1127570 Fix xball input parsing issue

### PR DESCRIPTION
**Pull request by dbenusov-xilinx**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1127570
xball does not work properly when asking the user for input. The tool will sometimes ask for input and ignore the given value always assuming a positive input was given.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
It seems that combining the python and shell script was causing strange issues with the input line parsing. This meant newlines were detected when they should not have been which caused the XRT tools to skip the user input and use the default positive response.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Split the xball shell and python into two seperate files and had the xball shell script invoke the python script with arguments.

#### Risks (if any) associated the changes in the commit
Possibility of unexpected behavior from the python script now that it is a standalone object. Some individuals may rely on the old behavior which was unpredictable. 

#### What has been tested and how, request additional testing if necessary
Multiple card manual input:
```
dbenusov@xsjsagarw50:/proj/rdi/staff/dbenusov/XRT$ xball xbutil reset
Discovering installed devices....


=====================================================================
1 / 4 [0000:b3:00.1] : xilinx_u200_gen3x16_xdma_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:b3:00.1 reset

Performing 'HOT Reset' on
  -[0000:b3:00.1]
Are you sure you wish to proceed? [Y/n]: n
Action canceled.

Command Return Value: 256 [Error(s) occured]


=====================================================================
2 / 4 [0000:65:00.1] : xilinx_vck5000_gen4x8_xdma_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:65:00.1 reset

Performing 'HOT Reset' on
  -[0000:65:00.1]
Are you sure you wish to proceed? [Y/n]: n
Action canceled.

Command Return Value: 256 [Error(s) occured]


=====================================================================
3 / 4 [0000:18:00.1] : xilinx_u30_gen3x4_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:18:00.1 reset

Performing 'HOT Reset' on
  -[0000:18:00.1]
Are you sure you wish to proceed? [Y/n]: n
Action canceled.

Command Return Value: 256 [Error(s) occured]


=====================================================================
4 / 4 [0000:17:00.1] : xilinx_u30_gen3x4_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:17:00.1 reset

Performing 'HOT Reset' on
  -[0000:17:00.1]
Are you sure you wish to proceed? [Y/n]: n
Action canceled.

Command Return Value: 256 [Error(s) occured]


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Summary:
   Installed device(s) : 4
          Shell Filter : '.'
      Number Evaluated : 4
                Passed : 0
                Failed : 4
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Multiple card --force flag:
```
dbenusov@xsjsagarw50:/proj/rdi/staff/dbenusov/XRT$ xball xbutil reset --force
Discovering installed devices....


=====================================================================
1 / 4 [0000:b3:00.1] : xilinx_u200_gen3x16_xdma_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:b3:00.1 reset --force

Performing 'HOT Reset' on
  -[0000:b3:00.1]
Are you sure you wish to proceed? [Y/n]: Y (Force override)
Successfully reset Device[0000:b3:00.1]

Command Return Value: 0 [Operation successful]


=====================================================================
2 / 4 [0000:65:00.1] : xilinx_vck5000_gen4x8_xdma_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:65:00.1 reset --force

Performing 'HOT Reset' on
  -[0000:65:00.1]
Are you sure you wish to proceed? [Y/n]: Y (Force override)
Successfully reset Device[0000:65:00.1]

Command Return Value: 0 [Operation successful]


=====================================================================
3 / 4 [0000:18:00.1] : xilinx_u30_gen3x4_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:18:00.1 reset --force

Performing 'HOT Reset' on
  -[0000:18:00.1]
Are you sure you wish to proceed? [Y/n]: Y (Force override)
ERROR: Reset relies on S/N, but S/N can't be read from SC: Invalid argument
Reset failed on Device[0000:18:00.1]

Command Return Value: 0 [Operation successful]


=====================================================================
4 / 4 [0000:17:00.1] : xilinx_u30_gen3x4_base_2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Command: /proj/xsjhdstaff6/dbenusov/XRT/build/Debug/opt/xilinx/xrt/bin/xbutil --device 0000:17:00.1 reset --force

Performing 'HOT Reset' on
  -[0000:17:00.1]
Are you sure you wish to proceed? [Y/n]: Y (Force override)
ERROR: Reset relies on S/N, but S/N can't be read from SC: Invalid argument
Reset failed on Device[0000:17:00.1]

Command Return Value: 0 [Operation successful]


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Summary:
   Installed device(s) : 4
          Shell Filter : '.'
      Number Evaluated : 4
                Passed : 4
                Failed : 0
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

#### Documentation impact (if any)
None

### CODE REVIEW

**Comment by salindac** [comment]
retest this please, infra error on the host

**Comment by dayeh-xilinx** [comment]
retest this please.

File: src/runtime_src/core/tools/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ if (${XRT_NATIVE_BUILD} STREQUAL "yes")
 endif()
 
 install (PROGRAMS "./common/xball" DESTINATION ${XRT_INSTALL_BIN_DIR})
+install (FILES "./common/xball.py" DESTINATION ${XRT_INSTALL_PYTHON_DIR})
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/tools/CMakeLists.txt:18]: in non-native build (edge build) , XRT_INSTALL_PYTHON_DIR is not defined. I think xball is not requried in edge case as there will be only one device 
You can move these under NATIVE_BUILD condition

Comment (dbenusov-xilinx) [reply src/runtime_src/core/tools/CMakeLists.txt:18]: That makes a lot of sense, I was wondering why it was failing on the pipeline! Thank you!

**Review by AShivangi — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/tools/common/xball.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import json
```
Comment (stsoe) [inline src/runtime_src/core/tools/common/xball.py:1]: Can we move this file under src/pyhton/tools?

Comment (dbenusov-xilinx) [reply src/runtime_src/core/tools/common/xball.py]: I ended up keeping the python code in the bash script as it originally was. Adding the python directory and file under yocto was a bit much just for this file

File: src/runtime_src/core/tools/common/xball [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 # ------------------------------------
 # 0. Prepare the execution environment
-
+""":" # Hide bash from python
```
Comment (keryell) [inline src/runtime_src/core/tools/common/xball:20]: OMG ;-)

**Review by rozumx — APPROVED** [review]

**Pull request merged**